libs/langchain/ragstack_langchain/graph_store/base.py

```diff
@@ -1,531 +1,12 @@
-            docsearch.as_retriever(
-                search_type="traversal",
-                search_kwargs={'k': 6, 'depth': 3}
-            )
-
-            # Retrieve more documents with higher diversity
-            # Useful if your dataset has many similar documents
-            docsearch.as_retriever(
-                search_type="mmr",
-                search_kwargs={'k': 6, 'lambda_mult': 0.25}
-            )
-
-            # Fetch more documents for the MMR algorithm to consider
-            # But only return the top 5
-            docsearch.as_retriever(
-                search_type="mmr",
-                search_kwargs={'k': 5, 'fetch_k': 50}
-            )
-
-            # Only retrieve documents that have a relevance score
-            # Above a certain threshold
-            docsearch.as_retriever(
-                search_type="similarity_score_threshold",
-                search_kwargs={'score_threshold': 0.8}
-            )
-
-            # Only get the single most similar document from the dataset
-            docsearch.as_retriever(search_kwargs={'k': 1})
-
-        """
-        return GraphStoreRetriever(vectorstore=self, **kwargs)
-
-
-class GraphStoreRetriever(VectorStoreRetriever):
-    """Retriever class for GraphStore."""
-
-    vectorstore: GraphStore
-    """GraphStore to use for retrieval."""
-    search_type: str = "traversal"
-    """Type of search to perform. Defaults to "traversal"."""
-    allowed_search_types: ClassVar[Collection[str]] = (
-        "similarity",
-        "similarity_score_threshold",
-        "mmr",
-        "traversal",
-        "mmr_traversal",
-    )
-
-    def _get_relevant_documents(
-        self, query: str, *, run_manager: CallbackManagerForRetrieverRun
-    ) -> List[Document]:
-        if self.search_type == "traversal":
-            return list(self.vectorstore.traversal_search(query, **self.search_kwargs))
-        if self.search_type == "mmr_traversal":
-            return list(
-                self.vectorstore.mmr_traversal_search(query, **self.search_kwargs)
-            )
-        return super()._get_relevant_documents(query, run_manager=run_manager)
-
-    async def _aget_relevant_documents(
-        self, query: str, *, run_manager: AsyncCallbackManagerForRetrieverRun
-    ) -> List[Document]:
-        if self.search_type == "traversal":
-            return [
-                doc
-                async for doc in self.vectorstore.atraversal_search(
-                    query, **self.search_kwargs
-                )
-            ]
-        if self.search_type == "mmr_traversal":
-            return [
-                doc
-                async for doc in self.vectorstore.ammr_traversal_search(
-                    query, **self.search_kwargs
-                )
-            ]
-        return await super()._aget_relevant_documents(query, run_manager=run_manager)
+__all__ = ["GraphStore", "GraphStoreRetriever", "Node", "nodes_to_documents"]
```

Comment:
I don't know that it is necessary to preserve backwards compatibility. I think it would be better to get to the new names sooner. To that end, we may want to change the names to the `GraphVector*` sooner rather than later. But, that can happen in a follow-up PR since it will likely need to change notebooks, etc.

Reply:
I think people who want to upgrade should not just change GraphStore to GraphVectorStore but use langchain instead of ragstack.
I'll update the notebooks once we have a release (since we are still usingthe extractors from ragstack-langchain and we need a compatible langchain version)

Reply:
I don't know if we need a deprecation warning system.
Probably not a lot of people are using this atm and this is not really needed. Also it's easy to upgrade.

Reply:
That makes sense. In the long term, do we see `ragstack_langchain/graph_store` going away, or do we think there will be utility in us providing extensions (eg., additional extractors, etc.) that are part of our package? If the latter, do we think it's worth always re-exporting things so the extensions can be used in one place? Or only exporting the extensions to emphasize that everyhting else comes from LangChain?

Reply:
Yes, in the long term, we can have components that stay into RAGStack.
But for the components that are in LangChain, I think we will eventually remove the reexports.